unity-client/Assets/Scripts/MainScripts/DCL/Controllers/CatalogController/CatalogController.cs

```diff
@@ -0,0 +1,345 @@
+
+                        break;
+                    }
+                default:
+                    {
+                        Debug.LogError("Bad type in item, will not be added to catalog");
+                        break;
+                    }
+            }
+        }
+
+        if (!string.IsNullOrEmpty(request.context))
+        {
+            ResolvePendingWearablesByContextPromise(request.context, request.wearables);
+            pendingWearablesByContextRequestedTimes.Remove(request.context);
+        }
+    }
+
+    public void WearablesRequestFailed(string payload)
+    {
+        WearablesRequestFailed requestFailedResponse = JsonUtility.FromJson<WearablesRequestFailed>(payload);
+
+        if (!string.IsNullOrEmpty(requestFailedResponse.context))
+        {
+            ResolvePendingWearablesByContextPromise(
+                requestFailedResponse.context,
+                null,
+                requestFailedResponse.error);
+        }
+        else
+        {
+            Debug.LogError(requestFailedResponse.error);
+        }
+    }
+
+    public void RemoveWearablesFromCatalog(string payload)
+    {
+        string[] itemIDs = JsonUtility.FromJson<string[]>(payload);
+
+        int count = itemIDs.Length;
+        for (int i = 0; i < count; ++i)
+        {
+            itemCatalog.Remove(itemIDs[i]);
+            wearableCatalog.Remove(itemIDs[i]);
+            wearablesInUseCounters.Remove(itemIDs[i]);
+        }
+    }
+
+    public void ClearWearableCatalog()
+    {
+        itemCatalog?.Clear();
+        wearableCatalog?.Clear();
+        wearablesInUseCounters.Clear();
+    }
+
+    public static Promise<WearableItem> RequestWearable(string wearableId)
+    {
+        Promise<WearableItem> promiseResult = new Promise<WearableItem>();
+
+        if (wearableCatalog.TryGetValue(wearableId, out WearableItem wearable))
+        {
+            wearablesInUseCounters[wearableId]++;
+            promiseResult.Resolve(wearable);
+        }
+        else
+        {
+            if (!awaitingWearablePromises.ContainsKey(wearableId))
+            {
+                awaitingWearablePromises.Add(wearableId, promiseResult);
+
+                // We accumulate all the requests during the same frames interval to send them all together
+                pendingRequestsToSend.Add(wearableId);
+            }
+            else
+            {
+                awaitingWearablePromises.TryGetValue(wearableId, out promiseResult);
+            }
+        }
+
+        return promiseResult;
+    }
+
+    public static Promise<WearableItem[]> RequestOwnedWearables(string userId)
+    {
+        Promise<WearableItem[]> promiseResult = new Promise<WearableItem[]>();
+
+        if (!awaitingWearablesByContextPromises.ContainsKey(OWNED_WEARABLES_CONTEXT))
+        {
+            awaitingWearablesByContextPromises.Add(OWNED_WEARABLES_CONTEXT, promiseResult);
```

Comment:
```suggestion
        Promise<WearableItem[]> promiseResult;

        if (!awaitingWearablesByContextPromises.ContainsKey(OWNED_WEARABLES_CONTEXT))
        {
            promiseResult = new Promise<WearableItem[]>();
            awaitingWearablesByContextPromises.Add(OWNED_WEARABLES_CONTEXT, promiseResult);
```
just to avoid an unneeded instantiation when promise is already in dictionary. wdyt?

Reply:
You're totally right! Thanks, Pato!

Reply:
Done!